### PR TITLE
Upgrade to BuildV3 and DeploymentV3

### PR DESCRIPTION
**Pull request by gwprice115**
Upgrade to use BuildV3 and DeploymentV3 to take advantage of the "global builds" feature

### CODE REVIEW

**Review by jared-gs — APPROVED** [review]

**Comment by gwprice115** [comment]
- need to be able to pass buildId as a flag to createBuild

File: internal/commands/build.go [hunk 1]
```diff
@@ -258,7 +257,7 @@ func buildFlagEnvVar(name string) string {
 var (
 	buildFlagEnvVarPrefix = globalFlagEnvVarPrefix + "BUILD_"
 
-	buildIDFlag = &workaround.IntFlag{
+	buildIDFlag = &cli.StringFlag{
```
Comment (natgabb) [inline internal/commands/build.go:260]: will this break with an int input or auto parse into string?

File: internal/commands/deploy.go [hunk 2]
```diff
@@ -149,6 +149,10 @@ func (c *DeployConfig) Merge(latest *shared.DeploymentV2, isIdleTimeoutDefault b
 func (c *DeployConfig) Validate() error {
 	var err error
 
+	if c.AppID == nil || *c.AppID == "" {
```
Comment (natgabb) [inline internal/commands/deploy.go:152]: so just confirming - removing this turns the input optional?

File: internal/commands/common.go [hunk 3]
```diff
@@ -110,10 +110,9 @@ func (c *GlobalConfig) Load(cmd *cli.Command) error {
 	if c.BaseURL == "" {
 		err = errors.Join(err, missingRequiredFlag(hathoraCloudEndpointFlag.Name))
 	}
-
 	appID := cmd.String(appIDFlag.Name)
 	if appID == "" {
-		err = errors.Join(err, missingRequiredFlag(appIDFlag.Name))
+		c.AppID = nil
```
Comment (natgabb) [inline internal/commands/common.go:115]: so just confirming - removing this turns the input optional?

Comment (gwprice115) [reply internal/commands/common.go:115]: yas

File: internal/commands/build.go (hunk 1 above)
Comment (gwprice115) [reply internal/commands/build.go:260]: auto parse into string

**Comment by gwprice115** [comment]
> * need to be able to pass buildId as a flag to createBuild and deploy commands

done

**Review by natgabb — APPROVED** [review]

**Pull request merged**